File: src/javaapp/MainMenu.java
package javaapp;

import java.util.Scanner;

public class MainMenu {

    private ProductCatalog productCatalog;

    public MainMenu(ProductCatalog productCatalog) {
        this.productCatalog = productCatalog;
    }

    public void displayMainMenu() {
        System.out.println("\nMain Menu :");
        System.out.println("1. Admin");
        System.out.println("2. Customer");
        System.out.println("0. Exit");
    }

    public void handleMainMenuInput() {
        System.out.print("Enter your choice : ");
        Scanner scanner = new Scanner(System.in);
        int choice = scanner.nextInt();
        switch (choice) {
            case 1:
                Admin admin = new Admin(productCatalog);
                AdminUi adminUi = new AdminUi(admin);
                while(true) {
                    adminUi.displayMenu();
                    adminUi.handleUserInput();
                }
            case 2:
                ShoppingCart shoppingCart = new ShoppingCart();
                ProductBrowser productBrowser = new ProductBrowser(productCatalog);
                Customer customer = new Customer(shoppingCart, productBrowser);
                CustomerUI customerUI = new CustomerUI(customer);
                while (true) {
                    customerUI.displayMenu();
                    customerUI.handleUserInput();
                }
            case 0:
                return;
            default:
                System.out.println("Invalid choice");
                break;
        }
    }
}

File: src/kotlinapp/MainMenu.kt
package kotlinapp

import javaapp.*
import javaapp.Admin
import javaapp.AdminUi
import javaapp.Customer
import javaapp.CustomerUI
import javaapp.ProductBrowser
import javaapp.ProductCatalog
import java.util.*

class MainMenu(private val productCatalog: ProductCatalog) {
    fun displayMainMenu() {
        println("\nMain Menu :")
        println("1. Admin")
        println("2. Customer")
        println("0. Exit")
    }

    fun handleMainMenuInput() {
        print("Enter your choice : ")
        val scanner = Scanner(System.`in`)
        val choice = scanner.nextInt()
        when (choice) {
            1 -> {
                val admin = Admin(productCatalog)
                val adminUi = AdminUi(admin)
                while (true) {
                    adminUi.displayMenu()
                    adminUi.handleUserInput()
                }
                val shoppingCart = ShoppingCart()
                val productBrowser = ProductBrowser(productCatalog)
                val customer = Customer(shoppingCart, productBrowser)
                val customerUI = CustomerUI(customer)
                while (true) {
                    customerUI.displayMenu()
                    customerUI.handleUserInput()
                }
                return
            }

            2 -> {
                val shoppingCart = ShoppingCart()
                val productBrowser = ProductBrowser(productCatalog)
                val customer = Customer(shoppingCart, productBrowser)
                val customerUI = CustomerUI(customer)
                while (true) {
                    customerUI.displayMenu()
                    customerUI.handleUserInput()
                }
                return
            }

            0 -> return
            else -> println("Invalid choice")
        }
    }
}


